panel/models/data.ts

```diff
@@ -14,13 +15,17 @@ export function transform_cds_to_records(cds: ColumnDataSource, addId: boolean =
       const shape = (array[0] == null || array[0].shape == null) ? null : array[0].shape;
       if ((shape != null) && (shape.length > 1) && (typeof shape[0] == "number"))
         item[column] = array.slice(i*shape[1], i*shape[1]+shape[1])
-      else
+      else if (array.length != cdsLength && (array.length % cdsLength === 0)) {
+        const offset = array.length / cdsLength
+	item[column] = array.slice(i*offset, i*offset+offset)
+      } else
         item[column] = array[i]
     }
     if (addId)
       item['_index'] = i
     data.push(item)
   }
+  console.log(data)
```

Comment:
Believe this one should be removed @philippjfr 😸

Reply:
It was.